lfroms/COMP4109-Project
# Saving messages and conversations to database

### What is the problem being solved in this PR?


Adding conversations and messages into the database

### Related issues or PRs


Resolves [issue 32]

### How did you accomplish this and why did you choose this approach?


Conversations are created with empty user and message fields. This will make it easy to add users in during creation.
Messages have to first search for the conversation with their Id first, and then they can create a new message to add to the database linking to the conversation
Also had to modify the message type so that it doesn't interfere with the socket

### How to test changes


Using pgAdmin 4 to see in database
- Navigate to localhost:3000/
- Create a new room with a user
- In pgAdmin, open up the query tool and use "select * from conversation"
- You should see the newly added conversation, and it's ID matches the params in your browser
- Send a message in the conversation
- In pgAdmin, query "select * from message"
- You should see the newly added message, and it's contents match the message in the browser

### Checklist
